src/shared/conf-parser.c

```diff
@@ -1398,21 +1398,48 @@ int config_parse_compression(
                 void *data,
                 void *userdata) {
 
-        Compression *compression = ASSERT_PTR(data);
+        CompressionOpts ***opts = ASSERT_PTR(data);
+        int level = -1;
+        char *l = NULL;
+        _cleanup_free_ char *t = NULL;
 
         if (isempty(rvalue)) {
-                *compression = COMPRESSION_NONE;
                 return 1;
         }
 
-        Compression c = compression_lowercase_from_string(rvalue);
+        l = strchr(rvalue, ':');
+        if (l) {
+                double lVal = strtol(l, NULL, 10);
```

Comment:
no, never use strtol directly. Impossible to use correctly, locale dependent and so on.

safe_atod()

Reply:
shouldn't this be `l + 1`?